app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :set_commentable
+
+  def create
+    @comment = @commentable.comments.new(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      set_comment_variables
+      respond_with_format(@commentable)
+    else
+      post_redirect_path(@commentable)
+    end
+  end
+
+  private
+
+  def set_commentable
+    @params = params[:comment]
+    id = @params[:commentable_id]
+    @commentable = if @params[:commentable_type] == 'Post'
+                     Post.find(id)
+                   else
+                     Comment.find(id)
+                   end
+  end
```

Comment:
You're not using `comment_params` method.
Also, the definition of `@params` as an instance variable seems to be not necessary

```suggestion
  def set_commentable
    @commentable = comment_params[:commentable_type].constantize.find(comment_params[:commentable_id])
  end
```